src/components/Clock.tsx

```diff
@@ -0,0 +1,54 @@
+import { Component } from 'react';
+
+type State = {
+  date: Date,
+};
+
+type Props = {
+  name: string,
+};
+
+export class Clock extends Component<Props, State> {
+  state = {
+    date: new Date(),
+  };
+
+  timerId = 0;
+
+  componentDidMount() {
+    this.timerId = window.setInterval(() => {
+      this.setState({ date: new Date() });
+    }, 1000);
+  }
+
+  componentDidUpdate(_: never, prevState: State) {
+    const { date } = this.state;
+
+    if (date !== prevState.date) {
+      window.console.info(date.toUTCString().slice(-12, -4));
+    }
```

Comment:
there's a better way of doing this in order to make your code more clear.
My suggestion is to use [Date.toLocaleString()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleString) method here